tensorflow/stream_executor/dnn.h

```diff
@@ -723,7 +741,9 @@ class AlgorithmConfig {
 
  private:
   AlgorithmType algorithm_;
+  bool          use_tensor_op_math_;
```

Comment:
We should consolite AlgorithmType and bool into a struct that represents the real algorithm to use. Either modify AlgorithmType, or add a new type. Either way is fine with me.

Reply:
Expanded AlgorithmType into a struct.